WeakAuras/Types.lua

```diff
@@ -1320,6 +1320,66 @@ for id, str in pairs(Private.combatlog_spell_school_types) do
   Private.combatlog_spell_school_types_for_ui[id] = ("%.3d - %s"):format(id, str)
 end
 
+Private.GetCurrencyListInfo = function(index)
+  if WeakAuras.IsRetail() then
+    return C_CurrencyInfo.GetCurrencyListInfo(index)
+  elseif WeakAuras.IsWrathClassic() then
+    local name, isHeader, isExpanded, isUnused, isWatched, _, icon, _, hasWeeklyLimit, _, _, itemID = GetCurrencyListInfo(index)
+    if itemID then
+      local _, currentAmount, _, earnedThisWeek, weeklyMax, totalMax, isDiscovered, rarity = GetCurrencyInfo(itemID)
+      local currencyInfo = {
+        name = name,
+        description = "",
+        isHeader = isHeader,
+        isHeaderExpanded = isExpanded,
+        isTypeUnused = isUnused,
+        isShowInBackpack = isWatched,
+        quantity = currentAmount,
+        trackedQuantity = 0,
+        iconFileID = icon,
+        maxQuantity = totalMax,
+        canEarnPerWeek = hasWeeklyLimit,
+        quantityEarnedThisWeek = earnedThisWeek,
+        isTradeable = false,
+        quality = rarity,
+        maxWeeklyQuantity = weeklyMax,
+        totalEarned = 0,
+        discovered = isDiscovered,
+        useTotalEarnedForMaxQty = false,
+      }
+      return currencyInfo
+    end
+  end
+end
+
+Private.GetCurrencyListSize = function()
```

Comment:
You can simplfy here by:
 if WeakAuras.IsRetail() then
     Private.GetCurrencyListSize = C_CurrencyInfo.GetCurrencyListSize()
  elseif WeakAuras.IsWrathClassic() then
    Private.GetCurrencyListSize = GetCurrencyListSize()
  end

Reply:
This results in errors on wrath and wrong behaviour on retail

Reply:
Huh, why? the API is exactly the same, you call with no arguments and get returned the size.

Reply:
```
if WeakAuras.IsRetail() then
  Private.GetCurrencyListSize = C_CurrencyInfo.GetCurrencyListSize()
elseif WeakAuras.IsWrathClassic() then
  Private.GetCurrencyListSize = GetCurrencyListSize()
end
```
`Private.CurrencyListSize` will always have the same value after this block. But the list size changes when there are collapsed headers and we expand/collapse them.

With `Private.CurrencyListSize()` the value will always be current.

Reply:
Eh right, typo in there:
```
if WeakAuras.IsRetail() then
  Private.GetCurrencyListSize = C_CurrencyInfo.GetCurrencyListSize
elseif WeakAuras.IsWrathClassic() then
  Private.GetCurrencyListSize = GetCurrencyListSize
end
```

Reply:
Yep, that's it. As a reward it spit out more bugs to fix 😁

Reply:
roo roo